frontend/src/components/AdminQuizComp/QuizList.tsx

```diff
@@ -1,33 +1,107 @@
 
-import Columns from "./QuizListColumns";
+// import Columns from "./QuizListColumns";
 import Table from '@components/UI/Table/Table';
 import Boxwidget from '@components/UI/Widget/Boxwidget/Boxwidget';
-import IQuizData from 'types/quiz.type';
+import Quiz from '../../models/quizModels';
 import Spinner from '@components/UI/Spinner/Spinner';
+import { ThunkDispatch } from 'redux-thunk';
+import { Action } from '../../store/actions/index'
+import { bindActionCreators } from 'redux';
+import { editQuizAdminModal } from '../../store/actions/action-creator/'
+import { connect } from 'react-redux'
+import {
+  PencilAltIcon,
+  MinusCircleIcon,
+  PlusCircleIcon
+} from "@heroicons/react/solid";
 
-interface Props {
+interface RProps {
   loading: boolean;
-  data: IQuizData[];
-  action: () => void;
+  data: Quiz[] | null;
 }
 
-interface Props {
-  action: () => void;
-}
 
-const QuizList: FC<Props> = ({ loading, data, action }) => {
+type Props = RProps & LinkDispatchProps;
+
+
+
+const QuizList: FC<Props> = ({ loading, data, editQuizAdminModal }) => {
+
+
+  const columns = [
+    {
+      Header: " ",
+      columns: [
+        {
+          Header: "Title",
+          accessor: "title",
+        },
+        {
+          Header: "Description",
+          accessor: "description",
+        },
+        {
+          width: 150,
+          Header: "Action",
+          Cell: (data: any) => {
+            let {
+              row: { original },
+            } = data;
+
+            return (
+              <>
+                <div className="w-full flex flex-end">
+                  <button
+                    //   onClick={() => props.openModal(original)}
+                    className="bg-gray-600 px-5 py-2 rounded-md text-white hover:bg-gray-800"
+                  >
+                    <PlusCircleIcon
+                      className="h-5 w-5 text-gray-400"
+                      aria-hidden="true" />
+                    {" "}
+                    Add Word
+                  </button>{" "}
+
+                  <button
+                    onClick={() => editQuizAdminModal(original)}
+                    className="bg-gray-600 px-5 py-2 rounded-md text-white hover:bg-gray-800 "
+                  >
+                    <PencilAltIcon
+                      className="h-5 w-5 text-gray-400"
+                      aria-hidden="true" />
+                    Edit
+                  </button>{" "}
+                  <button
+                    //   onClick={() => warningDeleting(original.u_id)}
+                    className="bg-red-700 p-2 rounded-md text-white hover:bg-red-800"
+                  >
+                    <MinusCircleIcon
+                      className="h-5 w-5 text-white"
+                      aria-hidden="true" />
+                    {" "}
+                    Delete
+                  </button>
+                </div>
+              </>
+            );
+          },
+        },
+      ],
+    },
+  ];

```

Comment:
refactor this one to make it readable and don't put large amount of html elements